### ISSUE TITLE
Visible fields are not saved correctly when filtering form fields

### ISSUE THREAD

**Issue by marvindurot** [wintercms/winter#950]
### Winter CMS Build

all

### PHP Version

8.1

### Database engine

MySQL/MariaDB

### Plugins installed

None

### Issue description

There is a bug in the Form widget : 
When you define a hidden field you want to show in the `filterFields()` model method, the later is not correctly saved.
This behavior occurs because the model form fields filtering is not applied in the `getSaveData()` widget method.

### Steps to replicate

1. Create a backend controller with its related model
2. Define a hidden field in the form fields definition
```yaml
# ===================================
#  Form Field Definitions
# ===================================

fields:
    value:
        label: Value
        hidden: true
        type: text
```
4. Define a `filterFields()` method in your test model that make your hidden field visible
```php
/**
 * Filtering form fields.
 *
 * @param object $fields
 * @param string $context
 */
 public function filterFields($fields, $context = null)
{
    $fields->value->hidden = false;
}
```
6. Trying to save the form in the backend

The value is not saved.

### Workaround

The workaround is to show your hidden fields in the form fields definition and hide them in your `filterFields` method.
That is to say, invert your visibility logic.

**Comment by mjauvin** [comment]
Should be fixed by #927

**Comment by marvindurot** [comment]
Nice :+1:  I was making the pull request when I saw it already existed!

**Comment by LukeTowers** [comment]
Fixed by #927
